rfcs/text/0000-matrix-service.md

```diff
@@ -0,0 +1,406 @@
+
+#### Update
+
+The Matrix Service does not respond to bundle updates in real-time as the service does not subscribe to DSS notifications. This is primarily due to the Matrix cache and the loss in performance when invalidating entries when the index is updated. The cache provides constant time delivery of matrices with respect to matrix size for previously requested matrices, however, not responding to bundle updates risks data consistency across DCP services and data availability to users.
+
+To balance these competing motivations, the proposed SOP being discussed with the DataOps team is that bundle updates will occur on a monthly basis in which the Matrix Service will respond to on the same cadence. While this poses a limitation on the rate of data updates through the system, the team proposes that the Matrix Service continue this patten and therefore guarantee support for DCP-wide data consistency for "Release Views" but not "Live Views".
+
+The Matrix Service also does not have a migration strategy for its index and requires a full re-index to respond to metadata schema changes.
+
+#### Delete/Redact
+
+Deletion and redaction are handled the same way and are straightforward processes for the Matrix Service:
+
+1. Remove all data from Redshift that is associated with bundle FQIDs from a given project UUID
+2. Delete project matrices from Data Brower's S3 bucket
+3. Clear the Matrix cache (see [Availability and Reproducibility](#availability-and-reproducibility) for details on the cache)
+
+This process is performed manually by a Matrix Service developer to ensure correct execution; it is anticipated that this process will occur infrequently. On project redaction, DataOps will open a Matrix Service ticket to request and track the process. The team adheres to a 1-week period to purge all associated data with redacted/deleted projects.
+
+### Availability
+
+#### Request ID
+
+The `request_id` returned by `POST /matrix` is valid for 30 days following the successful completion of the request. Specifically, this refers to the S3 download URL to retrieve the generated matrix will be valid for 30 days. Following this time, matrices are deleted from S3 and `GET /matrix/request_id` will notify clients that the request has expired.
+
+#### Matrix cache
+
+Matrices are cached in S3 according to a hash of a [request's input parameters](#post-/matrix). Specifically, the hash encodes the set of `cellkey`s selected by the user's `filter` object, the list of metadata `fields` exported, the output `format`, and `feature`. During a request, matching hashes short-circuit the matrix generation process, a new copy of the existing matrix is created for this request, and the request is completed. This enables constant time delivery with respect to matrix size for cache hits compared to linear time for cache misses.
```

Comment:
This should be `#post-matrix`.  github downcases, removes all punctuation, and then replaces ' ' with '-'.